lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,387 @@
+            cluster_name=cluster_name, esxi_host_name=esxi_host_name
+        )
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        self.results = {"before": {}, "after": {}, "changes": {}}
+
+    def sanitize_params(self, before, hostname, vmnic):
+        """
+        :before     : dict, of params on target interface before changing
+        :hostname   : str, hosthame
+        :vmnic      : srt, target nic
+        :return     : dict, of params forwarded to vCenter, or raise fail in case iconsistancy
+        """
+        params_to_change = {
+            "sriovEnabled": None,
+            "numVirtualFunction": None,
+            "msg": "",
+            "change": False,
+            "changes": {},
+        }
+        # change = False
+        if self.num_virt_func < 0:
+            self.module.fail_json(msg="allowed value for num_virt_func >= 0")
+
+        if self.num_virt_func == 0:
+            if self.sriov_on is True:
+                self.module.fail_json(
+                    msg="with sriov_on == true,  alowed value for num_virt_func > 0"
+                )
+            params_to_change.update({"sriovEnabled": False, "numVirtualFunction": 0})
+
+        if self.num_virt_func > 0:
+            if self.sriov_on is False:
+                self.module.fail_json(
+                    msg="with sriov_on == false,  alowed value for num_virt_func is 0"
+                )
+            params_to_change.update(
+                {"sriovEnabled": True, "numVirtualFunction": self.num_virt_func}
+            )
+
+            # check compatability
+            if not before["sriovCapable"]:
+                self.module.fail_json(
+                    msg="sriov not supported on host= %s, nic= %s" % (hostname, vmnic)
+                )
+
+            if (
+                before["maxVirtualFunctionSupported"]
+                < params_to_change["numVirtualFunction"]
+            ):
+                self.module.fail_json(
+                    msg="maxVirtualFunctionSupported= %d on %s"
+                    % (before["maxVirtualFunctionSupported"], vmnic)
+                )
+
+        # check current settings
+        if before["sriovEnabled"] != params_to_change["sriovEnabled"]:
+            params_to_change["changes"]["sriovEnabled"] = params_to_change[
+                "sriovEnabled"
+            ]
+            params_to_change["change"] = True
+        if before["numVirtualFunction"] != params_to_change["numVirtualFunction"]:
+            if (
+                before["numVirtualFunctionRequested"]
+                != params_to_change["numVirtualFunction"]
+            ):
+                params_to_change["changes"]["numVirtualFunction"] = params_to_change[
+                    "numVirtualFunction"
+                ]
+                params_to_change["change"] = True
+            else:
+                params_to_change["changes"]["msg"] = "Not active (looks not rebooted) "
```

Comment:
Not active (looks **like** not rebooted)?

Reply:
ok